src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -819,5 +832,37 @@ private static ILogger GetLogger(HttpContext httpContext)
                 return loggerFactory.CreateLogger(typeof(RequestDelegateFactory));
             }
         }
+
+        private static void EnsureRequestTaskOfNotNull<T>(Task<T> task)
+        {
+            if (task is null)
+            {
+                throw new InvalidOperationException("Task Result should not be null");
+            }
+        }
+
+        private static void EnsureRequestTaskNotNull(Task task)
+        {
+            if (task is null)
+            {
+                throw new InvalidOperationException("Task Result should not be null");
+            }
+        }
+
+        private static void EnsureRequestValueTaskOfNotNull<T>(ValueTask<T> task)
```

Comment:
Why do we need this check?

Reply:
It has been remove 🙂 . The check is directly done on the ValueTask<T> result with **`EnsureRequestResultNotNull`**